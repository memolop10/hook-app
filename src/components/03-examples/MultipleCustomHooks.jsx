import React from 'react'
import { useCounter } from '../../hooks/useCounter'
import useFetch from '../../hooks/useFetch'
import '../02-useEffect/effect.css'

const MultipleCustomHooks = () => {

    const { counter, increment } = useCounter(1);

    const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${ counter }`)
                              //true  
    const { author, quote } = !!data && data[0]
    console.log(data)
    console.log(author, quote)

    return (
        <div>
            <h1>BreakingBad</h1>
            <hr />

                {
                    loading ? (
                        <div className="alert alert-info text-center">
                            loading...
                        </div>
                    ): (
                        <blockquote className="blockquote text-right">
                            <p className="mb-2">{ quote }</p>
                            <footer className="blockquote-footer">{ author }</footer>
                        </blockquote>
                    )
                }

                <button 
                    className="btn btn-primary"
                    onClick={ increment }
                    > 
                    Next quote 
                </button>

        </div>
    )
}

export default MultipleCustomHooks
