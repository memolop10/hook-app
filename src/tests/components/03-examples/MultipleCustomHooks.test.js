import React from 'react'
import { shallow } from "enzyme";
import MultipleCustomHooks from "../../../components/03-examples/MultipleCustomHooks";
import { useCounter } from "../../../hooks/useCounter";
import useFetch from "../../../hooks/useFetch";

jest.mock( '../../../hooks/useFetch' );
jest.mock( '../../../hooks/useCounter' );



describe('Test in <MultipleCustomHooks/>', () => {

    useCounter.mockReturnValue({
        counter: 10,
        increment: () => {}
    });

    test('should render correctly', () => {
        
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks />)
        expect(wrapper).toMatchSnapshot()
    })

    test('should show the information', () => {

        useFetch.mockReturnValue({
            data: [{
                author: 'THE MEMO',
                quote: 'soy el mejor'
            }],
            loading: false,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks/>)
        // console.log(wrapper.html())

        expect( wrapper.find('.alert').exists() ).toBe( false );
        expect( wrapper.find('.mb-2').text().trim() ).toBe( 'soy el mejor' );
        expect( wrapper.find('footer').text().trim() ).toBe( 'THE MEMO' );
    })
    
    
});
