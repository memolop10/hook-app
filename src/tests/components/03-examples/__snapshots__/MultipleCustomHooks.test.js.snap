// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <MultipleCustomHooks/> should render correctly 1`] = `
<div>
  <h1>
    BreakingBad
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next quote
  </button>
</div>
`;
